collector-kinesis/src/main/java/zipkin/collector/kinesis/KinesisCollector.java

```diff
@@ -0,0 +1,155 @@
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.KinesisClientLibConfiguration;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorComponent;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.internal.Util;
+import zipkin.storage.StorageComponent;
+
+public final class KinesisCollector implements CollectorComponent, Closeable {
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static final class Builder implements CollectorComponent.Builder {
+
+    Collector.Builder delegate = Collector.builder(KinesisCollector.class);
+
+    AWSCredentialsProvider credentialsProvider;
+    String appName;
+    String streamName;
+
+    @Override
+    public Builder storage(StorageComponent storageComponent) {
+      delegate.storage(storageComponent);
+      return this;
+    }
+
+    @Override
+    public Builder metrics(CollectorMetrics collectorMetrics) {
+      delegate.metrics(Util.checkNotNull(collectorMetrics, "metrics").forTransport("kinesis"));
+      return this;
+    }
+
+    @Override
+    public Builder sampler(CollectorSampler collectorSampler) {
+      delegate.sampler(collectorSampler);
+      return this;
+    }
+
+    public Builder credentialsProvider(AWSCredentialsProvider credentialsProvider) {
+      this.credentialsProvider = credentialsProvider;
+      return this;
+    }
+
+    public Builder appName(String appName) {
+      this.appName = appName;
+      return this;
+    }
+
+    public Builder streamName(String streamName) {
+      this.streamName = streamName;
+      return this;
+    }
+
+    @Override
+    public KinesisCollector build() {
+      return new KinesisCollector(this);
+    }
+
+    Builder() {
+    }
+  }
+
+  private final Collector collector;
+  private final String appName;
+  private final String streamName;
+  private final AWSCredentialsProvider credentialsProvider;
+
+  private final Executor executor;
+  private Worker worker;
+  private IRecordProcessorFactory processor;
+
+  KinesisCollector(Builder builder) {
+    this.collector = builder.delegate.build();
+
+    this.appName = builder.appName;
+    this.streamName = builder.streamName;
+    this.credentialsProvider = builder.credentialsProvider;
+
+    executor = Executors.newSingleThreadExecutor(
+        new ThreadFactoryBuilder().setNameFormat("KinesisCollector-" + streamName + "-%d").build());
+  }
+
+  @Override
+  public KinesisCollector start() {
+    String workerId = null;
```

Comment:
comment why you need a random worker ID? ex in kafka I believe we intentionally choose the same worker ID in some situations.